crates/bevy_ui/src/render/mod.rs

```diff
@@ -1100,155 +1139,269 @@ pub fn prepare_uinodes(
+                                        ),
+                                        Vec2::new(
+                                            f32::min(clip.max.x - positions[2].x, 0.),
+                                            f32::min(clip.max.y - positions[2].y, 0.),
+                                        ),
+                                        Vec2::new(
+                                            f32::max(clip.min.x - positions[3].x, 0.),
+                                            f32::min(clip.max.y - positions[3].y, 0.),
+                                        ),
+                                    ]
+                                } else {
+                                    [Vec2::ZERO; 4]
+                                };
+
+                                let positions_clipped = [
+                                    positions[0] + positions_diff[0].extend(0.),
+                                    positions[1] + positions_diff[1].extend(0.),
+                                    positions[2] + positions_diff[2].extend(0.),
+                                    positions[3] + positions_diff[3].extend(0.),
+                                ];
+
+                                // cull nodes that are completely clipped
+                                let transformed_rect_size =
+                                    glyph.transform.transform_vector3(rect_size);
+                                if positions_diff[0].x - positions_diff[1].x
+                                    >= transformed_rect_size.x
+                                    || positions_diff[1].y - positions_diff[2].y
+                                        >= transformed_rect_size.y
+                                {
+                                    continue;
+                                }
+
+                                let uvs = [
+                                    Vec2::new(
+                                        glyph.rect.min.x + positions_diff[0].x,
+                                        glyph.rect.min.y + positions_diff[0].y,
+                                    ),
+                                    Vec2::new(
+                                        glyph.rect.max.x + positions_diff[1].x,
+                                        glyph.rect.min.y + positions_diff[1].y,
+                                    ),
+                                    Vec2::new(
+                                        glyph.rect.max.x + positions_diff[2].x,
+                                        glyph.rect.max.y + positions_diff[2].y,
+                                    ),
+                                    Vec2::new(
+                                        glyph.rect.min.x + positions_diff[3].x,
+                                        glyph.rect.max.y + positions_diff[3].y,
+                                    ),
+                                ]
+                                .map(|pos| pos / atlas_extent);
```

Comment:
Some of these calculations are duplicated in the Node / Glyph cases. It's not a big issue but maybe they could be moved to a small helper function or closure? This would make the code less verbose and ensure that they are equivalent if they change later.